### ISSUE TITLE
Preserve GMT offset

### ISSUE THREAD

**Issue by damianooldoni** [inbo/camtrapdp#204]
From @MarcusRowcliffe mail, Nov 2, 2022:

> I believe importing data using read_cameratrap_dp sets observations$timestamp time zone to UTC, adjusting the time according to the GMT offset at the end of the original character timestamp values. Is that GMT offset value preserved anywhere in the data?

My answer via mail for reference:

> the function uses the function [read_resource()](https://docs.ropensci.org/frictionless/reference/read_resource.html) from the frictionless package behind the hood with default arguments: see [line of code](https://github.com/inbo/camtraptor/blob/822e06f5d9768d5e431b53ce329723e2acd28032/R/read_camtrap_dp.R#L78) for deployments rsource as example. However, read_resource() supports several [field types](https://docs.ropensci.org/frictionless/reference/read_resource.html#field-types) such as timestamps with timezones (so other than UTC timestamps). As read_camtrap_dp doesn't expose this field types to the user it seems indeed that only UTC is supported now.

@MarcusRowcliffe: In [example](https://github.com/tdwg/camtrap-dp/blob/main/example/deployments.csv) of camtrap-dp I can indeed see how GMT offset is part of the datetime, e.g. `2020-05-30T04:57:37+02:00`. I will work to allow importing these datetimes as they are in the data package next week! Thanks for reporting this and sorry for the delay.

**Comment by damianooldoni** [comment]
@peterdesmet: I think this is a camtrapdp related issue, or a frictionless one as camtrapdp doesn't do anything more than using frictionless::read_resource() multiple times, see [#L100-L105](https://github.com/inbo/camtrapdp/blob/ca33a53433de9ef7d77572d4a28783523c481e46/R/read_camtrapdp.R#L100-L105).

Shortly, based on the example: `example_dataset()` reads data from [`camtrap-dp/example`](https://github.com/tdwg/camtrap-dp/tree/1.0.2/example). In this example the datetime columns have a timezone (GMT offset). However, the camtrapdp function `read_camtrapdp()` convert automatically to UTC.

**Comment by peterdesmet** [comment]
> However, the camtrapdp function `read_camtrapdp()` convert automatically to UTC.

That is correct. The offset is correctly interpreted _AND_ the entire timestamp is converted to UTC. This is a feature of readr (used by frictionless):

``` r
library(readr)
input <- "2020-05-30T00:00:00+02:00"
output <- readr::parse_datetime(input)
output
#> [1] "2020-05-29 22:00:00 UTC"
class(output)
#> [1] "POSIXct" "POSIXt"
```

<sup>Created on 2026-02-17 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

As a result, we can't retain the original offset. Doing so would require major development. I would venture that for analysis it's beneficial to have everything in UTC.

**Comment by damianooldoni** [comment]
Thanks, @peterdesmet.
@MarcusRowcliffe: users interested in express times using local time (with GMT offset), maybe they could use the [{timezone}](https://github.com/rundel/timezone) R package? Is it ok to leave this functionality out of the package?
